### DISCUSSION TITLE
How can I remove empty space in end line

### DISCUSSION THREAD

**Discussion by simix-alessandrokantousian** [Cinchoo/ChoETL#275] [Q&A] [answered]
Hi Guys, I'm using ChoETL library to read some files and I have a problem with txt files, When I read the lines in a file with 265 lines, about 10 lines it is reading correctly, but the others are shown only null and when I change white space end line  in one of the 10 lines correctly, the program return null in this line too. I'd like to know How Can I remove Empty Space in end line using **ChoFixedLengthRecordFieldConfiguration** or **ChoFixedLengthReader**?

**Comment by Cinchoo** [comment]
pls write sample fiddle at https://dotnetfiddle.net/ and explain more in detail, will try to help u with the issue.

**Reply by simix-alessandrokantousian** [reply]
Thank for your attetion, follow my code https://dotnetfiddle.net/0GmH0e 

When I have two lines that have the same length, it works, but when these lines have different lengths, it doesn't work, look this image:

![image](https://user-images.githubusercontent.com/91629430/221013028-e9788a76-6946-433f-84ac-f8fa72270176.png)
![image](https://user-images.githubusercontent.com/91629430/221013080-d7dcee52-3864-4fe2-99b5-c8424f8fbeff.png)

It is possible to solve this using ChoETL?

**Comment by Cinchoo** [comment, marked as answer]
Well, currently I doesn't allow variable record length. But you can solve by adding `OnBeforeRecordField` event and adjust the line. Here is the working sample

https://dotnetfiddle.net/niuJEi

FYI, in the fixed length record file parsing, all fields must be specified with `startindex` and `length`. Otherwise, results will be unexpected.

**Reply by simix-alessandrokantousian** [reply]
I would like to thank you for your attention and help, we saw your update regarding our case that we asked for help and we have already downloaded it :grin: :tada:

**Comment by simix-markus** [comment]
Hi, I work with Alessandro.
The data has fixed lenght (kinda of), but each line has ramdom blank spaces after, outside of our control because is generated by other software.
Thanks! This solve the issue.
